From 88a2cf6c4bab6e94a65e9c0db8813709372e9180 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Wed, 19 Aug 2020 13:47:37 +0000
Subject: [PATCH 030/267] Add new STATX_* constants from Linux 5.8 to
 bits/statx-generic.h.

This patch adds the new STATX_MNT_ID, STATX_ATTR_MOUNT_ROOT and
STATX_ATTR_DAX macros from Linux 5.8 to glibc's bits/statx-generic.h.
(As with previous such changes, this only does anything if glibc is
being used with old kernel headers.)

A comment in the Linux kernel headers indicates that STATX_ALL is
deliberately not being changed.

Tested for x86_64.
---
 io/bits/statx-generic.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/io/bits/statx-generic.h b/io/bits/statx-generic.h
index 7a09e94be2..c34697e3c1 100644
--- a/io/bits/statx-generic.h
+++ b/io/bits/statx-generic.h
@@ -40,6 +40,7 @@
 # define STATX_BASIC_STATS 0x07ffU
 # define STATX_ALL 0x0fffU
 # define STATX_BTIME 0x0800U
+# define STATX_MNT_ID 0x1000U
 # define STATX__RESERVED 0x80000000U
 
 # define STATX_ATTR_COMPRESSED 0x0004
@@ -48,7 +49,9 @@
 # define STATX_ATTR_NODUMP 0x0040
 # define STATX_ATTR_ENCRYPTED 0x0800
 # define STATX_ATTR_AUTOMOUNT 0x1000
+# define STATX_ATTR_MOUNT_ROOT 0x2000
 # define STATX_ATTR_VERITY 0x100000
+# define STATX_ATTR_DAX 0x2000
 #endif /* !STATX_TYPE */
 
 __BEGIN_DECLS
-- 
2.29.0

